simple/operator.go

```diff
@@ -195,6 +202,9 @@ func (o *Operator) WatchKind(kind resource.Kind, watcher SyncWatcher, options Li
 		return err
 	}
 	ow, err := operator.NewOpinionatedWatcherWithFinalizer(kind, client, func(sch resource.Schema) string {
+		if o.FinalizerGenerator != nil {
+			return o.FinalizerGenerator(sch)
+		}
 		if o.Name != "" {
 			return fmt.Sprintf("%s-%s-finalizer", o.Name, kind.Plural())
```

Comment:
I would also trim this to just use the name. The kind is already clear (because that's what we're attaching the finalizer to) and `-finalizer` is clear because it's part of `metadata.finalizers`.
```suggestion
			return o.Name
```

An example from built-in Namespace finalizer:
```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: reloader
spec:
  finalizers:
  - kubernetes
status:
  phase: Active
```

Reply:
I don't want to change this at the moment without first confirming what existing apps are using--otherwise it may change their finalizers and break operator function. But agreed that we can shorten the defaults once we get the ability to customize the finalizer name into `simple.Operator`.